services/QuillLMS/spec/controllers/concerns/teacher_fixes_spec.rb

```diff
@@ -394,5 +394,61 @@
         end
       end
     end
+
+  end
+
+  describe '#recover_classroom_units_and_associated_activity_sessions' do
+    it 'should unarchive the classroom units' do
+      classroom_unit_with_activity_sessions1.update(visible: false)
+      classroom_unit_with_activity_sessions2.update(visible: false)
+      TeacherFixes::recover_classroom_units_and_associated_activity_sessions([classroom_unit_with_activity_sessions1, classroom_unit_with_activity_sessions2])
+      expect(classroom_unit_with_activity_sessions1.reload.visible).to eq(true)
+      expect(classroom_unit_with_activity_sessions2.reload.visible).to eq(true)
+    end
+
+    it 'should unarchive the associated activity sessions' do
+      classroom_units = [classroom_unit_with_activity_sessions1, classroom_unit_with_activity_sessions2]
+      activity_sessions = ActivitySession.where(classroom_unit: classroom_units)
+      activity_sessions.each { |as| as.update(visible: false) }
+      TeacherFixes::recover_classroom_units_and_associated_activity_sessions(classroom_units)
+      activity_sessions.each { |as| expect(as.reload.visible).to eq(true) }
+    end
+
+    it 'should touch each activity session to bubble up touch for cache invalidation purposes' do
+      first_activity_session = ActivitySession.find_by(classroom_unit: classroom_unit_with_activity_sessions1)
+      stored_updated_at_value = first_activity_session.updated_at
+      TeacherFixes::recover_classroom_units_and_associated_activity_sessions([classroom_unit_with_activity_sessions1])
+      expect(first_activity_session.reload.updated_at).not_to equal(stored_updated_at_value)
```

Comment:
There's nothing wrong with this formulation, but another way to write this spec is with an `expect... to change` block:
```
expect do
  TeacherFixes::recover_classroom_units_and_associated_activity_sessions([classroom_unit_with_activity_sessions1])
end.to change(ActivitySession.find_by(classroom_unit: classroom_unit_with_activity_sessions1), :updated_at)
```
I don't see a reason to change what you've got, but figured I'd call out the alternate approach for more complex situations in the future.